### PR TITLE
Allow server path to be prefixed

### PR DESCRIPTION
**Pull request by chrmod**
In cases when `tiny-lr` is being used as middleware together with other applications, there is a change for routes collision (for example other app can respond to `GET /alert`).

This PR introduces a `prefix` options for a server, that alters all routes with the prefix.

This PR is going to simplify ember-cli changes introduced in this PR https://github.com/ember-cli/ember-cli/pull/7508

TODO:
This depends on the changes in `livereload-js`, need to release them first, before merging this PR.
https://github.com/livereload/livereload-js/pull/66

### CODE REVIEW

**Comment by stefanpenner** [comment]
@elwayman02 / @mklabs got any cycles to take a look at ^ ?

**Comment by chrmod** [comment]
@elwayman02 @mklabs @stefanpenner have figure out that passing specific options to `livereload-js` is a hosting project responsibility, in `ember-cli` case it is set by `ember-cli-inject-live-reload` ember addon.
It works like that as options are passed as parameters to the `livereload.js` GET request, https://github.com/ember-cli/ember-cli-inject-live-reload/pull/45/files#diff-168726dbe96b3ce427e7fedce31bb0bcR31

File: test/server.js [hunk 1]
```diff
@@ -2,31 +2,31 @@ import request from 'supertest';
 import assert from 'assert';
 import listen from './helpers/listen';
 
-describe('tiny-lr', () => {
```
Comment (elwayman02) [inline test/server.js]: Should there not still be a top-level `describe` here for the test suite?

**Comment by elwayman02** [comment]
This seems pretty sane to me. Can you add documentation for this to the README? Otherwise it seems good to go!

**Comment by chrmod** [comment]
@elwayman02 I've updated the readme and `livereload-js` to support http server route prefixes. Had to publish a fork of `livereload-js` as the project is not maintained. 

@stefanpenner If you prefer, we can publish `livereload-js` under `ember-cli` npm organization, but then you will have to fork the project under github organization - don't think I can do that with current permissions.

**Comment by elwayman02** [comment]
@stefanpenner let me know how you'd like to move forward regarding the livereload-js fork. If y'all could help resolve #93, which has been waiting for an open PR in livereload-js for about 2 years, that would be helpful to roll this all into the new releases.

**Comment by elwayman02** [comment]
@stefanpenner @chrmod It looks like there's a new maintainer looking at PRs in `livereload-js` today. Perhaps we can avoid the fork if we can get their attention on what we need while they're around?

**Comment by chrmod** [comment]
@elwayman02 livereload-js has relased 2.3, so I've updated the dependency. Would be great to have this one in.

**Comment by elwayman02** [comment]
@chrmod looks like it's still searching for the fork somewhere:

> Cannot find module '@chrmod/livereload-js/dist/livereload.js'

**Comment by chrmod** [comment]
@elwayman02 sorry about that. should be fine now

**Pull request merged**